lib/Pipeline/Step.cpp

```diff
@@ -494,6 +494,8 @@ Step::storeInvalidationMetadata(const revng::DirectoryPath &Path) const {
 
     ZstdCompressedOstream OS(File->get()->os(), 5);
     ::serialize(OS, ToStore);
+    OS.flush();
```

Comment:
> Without flushing the output, the invalidation metadata could end up being half-written

How? In case of a crash?

The next few lines don't seem to be crash-prone.

Reply:
> How? In case of a crash?

No, it's a bit subtle, basically the order of events is:
1. Get `WritableFile`, which has a `llvm::raw_fd_ostream`
2. Wrap ostream in `ZstdCompressedOstream`
3. Serialize stuff on the zstdOstream
4. Run `WritableFile.commit`. It calls the `raw_fd_ostream.flush` but does not flush the wrapping zstdOstream, so the last block of zstd data is never written to the file.
5. The storage stuff takes the partially written file and does its business, unaware that the file is not complete

Reply:
OK, usually in these cases I wrap `OS` into a scope just to force triggering the destructor.

It's not necessarily better, just a pattern we already adopt around the code base.

Reply:
Switched to the scope.